File: abc/085/b.cpp
#define _USE_MATH_DEFINES  // M_PI等のフラグ

#include <iostream>
#include <string>
#include <cstring>
#include <cstdlib>
#include <cmath>
#include <algorithm>
#include <map>
#include <vector>
#include <list>

#define MOD 1000000007
#define COUNTOF(array) (sizeof(array)/sizeof(array[0]))

using namespace std;


void solve(){
    int N;
    cin >> N;
    vector<int> Ds(N);
    for (int i=0; i<N; i++)
    {
        cin >> Ds.at(i);
    }

    sort(Ds.begin(), Ds.end());
    reverse(Ds.begin(), Ds.end());

    int ans = 1;
    int pre_d = Ds.at(0);
    for (int i=1; i<N; i++)
    {
        if (Ds.at(i) < pre_d)
        {
            ans ++;
            pre_d = Ds.at(i);
        }
    }

    cout << ans << endl;
}

int main(int argc, char const *argv[]){
    solve();
    return 0;
}

File: abc/085/b.py
# -*- coding:utf-8 -*-

def solve():
    N = int(input())
    D = []
    for i in range(N):
        D.append(int(input()))
    D.sort()
    
    pre_d = D[0]
    ans = 1
    for i in range(1, N):
        if pre_d < D[i]:
            pre_d = D[i]
            ans += 1
    print(ans)
    

if __name__ == "__main__":
    solve()
